### DISCUSSION TITLE
3 Problems Encountered When Using nicegui in Jupyterlab

### DISCUSSION THREAD

**Discussion by clueple** [zauberzeug/nicegui#1777] [Q&A]
### Description

NiceGUI 1.3.16 is great and i tried the auto reload feature of nicegui on Jupyterlab (for the test code below)

```py
from nicegui import ui

input = ui.input(label='input123')

ui.run()
```
3 'problems' noted:
1)  nicegui not able to detect python file save within jupyterlab
2) nicegui requires shutdown of all kernels (and restart) to be able to auto reload (and it works only once)
3) when I tried shutting down jupyterlab, it was frozen (not able to log out / shutdown jupyterlab except hard close the window)

It's quite hard to explain in words and I show it in a video

https://youtu.be/pxPs8Xtap00?si=KE26m_18a9QdW0bj

It would be nice if nicegui works in jupyterlab (especially for live tutorial).

Many thanks for any help.

**Comment by rodja** [comment]
Have you seen https://github.com/zauberzeug/nicegui/discussions/733? It's a bit hidden and we would love to see it in an official example but so far we lack volunteers...

**Comment by clueple** [comment]
> Have you seen #733? It's a bit hidden and we would love to see it in an official example but so far we lack volunteers...

Thanks for the post.  I was searching 'nicegui jupyterlab' before posting my issue but couldn't find one.
I'm trying to learn how the threading approach works.  

Thanks.
